lib/relationships.js

```diff
@@ -0,0 +1,123 @@
+'use strict';
+
+var _ = require('lodash');
+var utils = require('./utils');
+
+module.exports = function (app, options) {
+  //get remote methods.
+  //set strong-remoting for more information
+  //https://github.com/strongloop/strong-remoting
+  var remotes = app.remotes();
+  var id, data, model;
+
+  remotes.before('**', function (ctx, next) {
+
+    id = ctx.req.params.id;
+    data = ctx.args.data;
+    model = utils.getModelFromContext(ctx, app);
+    relationships(id, data, model);
+
+    next();
+  });
+
+  // for create
+  remotes.after('**', function (ctx, next) {
+
+    if (ctx.result) {
+      id = ctx.result.id;
+      data = ctx.req.body;
+      model = utils.getModelFromContext(ctx, app);
+      relationships(id, data, model);
+    };
+
+    next();
+  });
+
+};
+
+function relationships (id, data, model) {
+
+  if (!data || !data.data || !id || !model || !data.data.relationships) {
+    return;
+  }
+
+  _.each(data.data.relationships, function (relationship, name) {
+
+    var serverRelation = model.relations[name];
+    var type = serverRelation.type;
+    var modelTo = serverRelation.modelTo;
+
+    var fkName = serverRelation.keyTo;
+
+    if (type === 'belongsTo') {
+      fkName = serverRelation.keyFrom;
+      modelTo = serverRelation.modelFrom;
+    }
+
+    if (!modelTo) {
+      return false;
+    }
+
+    var setTo = {};
+    setTo[fkName] = null;
+    var where = {};
+    where[fkName] = id;
+
+    // remove all relations to the model (eg .: post/1)
+    if (type !== 'belongsTo') {
+      modelTo.updateAll(where, setTo, function (err, info) {
```

Comment:
It might also work to move this code inside the 
`if (_.isArray(relationship.data)) {` instead of checking for belongsTo